File: matlab/src/api/mam/mfq_sojourn.m
%{ @file mfq_sojourn.m
 %  @brief Sojourn-time distribution of a Markov-modulated fluid queue
 %
 %  @author LINE Development Team
%}

%{
 % @brief Sojourn-time distribution of a fluid queue as an ME/PH representation.
 %
 % @details
 % Thin LINE wrapper around FluidQueueSTD. Returns the distribution of the time
 % a fluid drop spends in a Markov-modulated fluid queue with input rate matrix
 % Rin and output (service) rate matrix Rout, modulated by generator Q, as a
 % matrix-exponential (ME) or phase-type (PH) representation (alpha, A).
 %
 % @par Syntax:
 % @code
 % [alpha,A] = mfq_sojourn(Q,Rin,Rout)
 % [alpha,A] = mfq_sojourn(Q,Rin,Rout,Q0,transToPH)
 % @endcode
 %
 % @par Parameters:
 % <table>
 % <tr><th>Name<th>Description
 % <tr><td>Q<td>(N,N) generator of the background Markov chain
 % <tr><td>Rin<td>(N,N) diagonal input fluid-rate matrix
 % <tr><td>Rout<td>(N,N) diagonal output (service) fluid-rate matrix
 % <tr><td>Q0<td>(Optional) level-0 generator (default: Q)
 % <tr><td>transToPH<td>(Optional) true to return a PH (else ME) representation
 % </table>
 %
 % @par Returns:
 % alpha, A: ME/PH representation of the sojourn-time distribution.
%}
function [alpha,A] = mfq_sojourn(Q,Rin,Rout,Q0,transToPH)
if ~exist('Q0','var'), Q0 = []; end
if ~exist('transToPH','var'), transToPH = false; end
[alpha,A] = FluidQueueSTD(Q,Rin,Rout,Q0,transToPH);
end

File: jar/src/main/java/jline/api/mam/Mfq_sojourn.java
/**
 * @file Sojourn-time distribution of a Markov-modulated fluid queue
 * @since LINE 3.0
 */
package jline.api.mam;

import jline.lib.butools.queues.FluidQueueSTD;
import jline.util.matrix.Matrix;

public final class Mfq_sojourn {
    private Mfq_sojourn() {}

    /** Sojourn-time distribution as {alpha, A} (ME if transToPH=false, PH otherwise). */
    public static Matrix[] mfq_sojourn(Matrix Q, Matrix Rin, Matrix Rout, Matrix Q0, boolean transToPH) {
        return FluidQueueSTD.fluidQueueSTD(Q, Rin, Rout, Q0, transToPH);
    }
}
